collection-helpers_tests.js

```diff
@@ -1,60 +1,130 @@
-Tinytest.add("works", function(test) {
-  Books = new Mongo.Collection('books' + test.id);
-  Authors = new Mongo.Collection('authors' + test.id);
+const meteorVersion = Meteor.release ? parseFloat(Meteor.release.split('@')[1]) : null;
 
-  var author1 = Authors.insert({
-    firstName: 'Charles',
-    lastName: 'Darwin'
-  });
+if (meteorVersion && meteorVersion >= 3.0) {
+  Tinytest.add("works", async function (test) {
+    Books = new Mongo.Collection('books' + test.id);
+    Authors = new Mongo.Collection('authors' + test.id);
 
-  var author2 = Authors.insert({
-    firstName: 'Carl',
-    lastName: 'Sagan'
-  });
+    var author1 = await Authors.insertAsync({
+      firstName: 'Charles',
+      lastName: 'Darwin'
+    });
 
-  var book1 = Books.insert({
-    authorId: author1,
-    name: 'On the Origin of Species'
-  });
+    var author2 = await Authors.insertAsync({
+      firstName: 'Carl',
+      lastName: 'Sagan'
+    });
 
-  var book2 = Books.insert({
-    authorId: author2,
-    name: 'Contact'
-  });
+    var book1 = await Books.insertAsync({
+      authorId: author1,
+      name: 'On the Origin of Species'
+    });
 
-  Books.helpers({
-    author: function() {
-      return Authors.findOne(this.authorId);
-    }
-  });
+    var book2 = await Books.insertAsync({
+      authorId: author2,
+      name: 'Contact'
+    });
 
-  // We should be able to apply more if we wish
-  Books.helpers({
-    foo: 'bar'
-  });
+    Books.helpers({
+      author: function () {
+        return Authors.findOne(this.authorId);
+      },
+
+      authorAsync: async function() {
+        return await Authors.findOneAsync(this.authorId)
+      }
+    });
+
+    // We should be able to apply more if we wish
+    Books.helpers({
+      foo: 'bar'
+    });
+
+    Authors.helpers({
+      fullName: function () {
+        return this.firstName + ' ' + this.lastName;
+      },
+      books: function () {
+        return Books.find({ authorId: this._id });
+      }
+    });
+
+    var book = await Books.findOneAsync(book1);
+    var author = book.author();
+    test.equal(author.firstName, 'Charles');
+    test.equal(book.foo, 'bar');
 
-  Authors.helpers({
-    fullName: function() {
-      return this.firstName + ' ' + this.lastName;
-    },
-    books: function() {
-      return Books.find({ authorId: this._id });
-    }
+    book = await Books.findOneAsync(book2);
+    author = book.author();
+    test.equal(author.fullName(), 'Carl Sagan');
+
+    author = await book.authorAsync();
+    test.equal(author.fullName(), 'Carl Sagan');
+
+    author = await Authors.findOneAsync(author1);
+    books = author.books();
+    test.equal(books.count(), 1);
```

Comment:
Shouldn't this be failing since the helper is using the non async `Books.find`?

Reply:
On the client no, but on the server, yes. Good catch. I'm guessing the returned `Promise` autocasts to a truthy value and so the comparison to ``1`` passes? I'll take a look.

Reply:
Holllyyy cow, so if I put:
```
test.isTrue('x' === 'y')
```
in the Meteor 3+ tests and all tests pass, so long as I put that line AFTER the first 'await'.

I did a bit of digging and Tinytest does NOT play nice with `async`.

I'm going to look at replacing the tests with Mocha, I guess? Agg, kill me.